pallets/nomad/da-bridge/src/message.rs

```diff
@@ -0,0 +1,109 @@
+use nomad_core::{TypedMessage, TypedMessageVariant};
+use primitive_types::H256;
+use sp_std::vec::Vec;
+
+/// DA Bridge message 1-byte type tags. Note that the invalid 0 variant
+/// is to maintain parity with our existing Solidity contracts.
+#[repr(u8)]
+#[derive(Clone, Debug, PartialEq)]
+pub enum DABridgeMessageTypes {
+	Invalid = 0,
+	DataRootMessage = 1,
+}
+
+impl From<u8> for DABridgeMessageTypes {
+	fn from(num: u8) -> Self {
+		match num {
+			0 => Self::Invalid,
+			1 => Self::DataRootMessage,
+			_ => panic!("Invalid u8 for DABridgeMessage enum!"),
+		}
+	}
+}
+
+/// Data root message variant. Contains a block number and its corresponding
+/// data root.
+#[derive(Clone, Debug, PartialEq)]
+pub struct DataRootMessage {
+	/// Block number
+	pub block_number: u32,
```

Comment:
`block_number` should be a generic type, like `frame_system::Config::BlockNumber`.

Reply:
The receiving solidity contracts expect a 4-byte u32 for block number. We can have this field be generic but ultimately the Solidity contract needs to match the block number size so thought it would make sense to just have it be concrete. Thoughts?